packages/api/src/external/ehr/elation/command/process-patients-from-appointments.ts

```diff
@@ -0,0 +1,255 @@
+    ...new Map(
+      patientAppointmentsFromGetAppointmentsByPractice.map(app => [app.elationPatientId, app])
+    ).values(),
+  ];
+  const patientAppointmentsUniqueByPractice: { [k: string]: PatientAppointment[] } = {};
+  patientAppointmentsUnique.map(appointment => {
+    const cxId = appointment.cxId;
+    const practiceId = appointment.elationPracticeId;
+    const key = createMapKey(cxId, practiceId);
+    if (patientAppointmentsUniqueByPractice[key]) {
+      patientAppointmentsUniqueByPractice[key].push(appointment);
+    } else {
+      patientAppointmentsUniqueByPractice[key] = [appointment];
+    }
+  });
+
+  const errorsFromGetPatientIdOrFailByPractice: string[] = [];
+  const argsForGetPatientIdOrFailByPractice = Object.keys(patientAppointmentsUniqueByPractice).map(
+    key => {
+      const { cxId, practiceId } = parseMapKey(key);
+      return {
+        cxId,
+        practiceId,
+        environment: elationEnvironment,
+        patientAppointmentsUnique: patientAppointmentsUniqueByPractice[key] ?? [],
+        errorArray: errorsFromGetPatientIdOrFailByPractice,
+        log,
+      };
+    }
+  );
+
+  await executeAsynchronously(argsForGetPatientIdOrFailByPractice, getPatientIdOrFailByPractice, {
+    numberOfParallelExecutions: parallelPractices,
+    delay: delayBetweenPracticeBatches.asMilliseconds(),
+  });
+
+  if (errorsFromGetPatientIdOrFailByPractice.length > 0) {
+    capture.error("Failed to find or create patients", {
+      extra: {
+        getPatientIdOrFailArgsCount: patientAppointmentsUnique.length,
+        errorCount: errorsFromGetPatientIdOrFailByPractice.length,
+        errors: errorsFromGetPatientIdOrFailByPractice.join(","),
+        context: "elation.get-patients-from-appointments",
+      },
+    });
+  }
+}
+
+async function getAppointmentsByPractice({
+  cxId,
+  practiceId,
+  environment,
+  fromDate,
+  toDate,
+  returnArray,
+  errorArray,
+  log,
+}: {
+  cxId: string;
+  practiceId: string;
+  environment: ElationEnv;
+  fromDate: Date;
+  toDate: Date;
+  returnArray: PatientAppointment[];
+  errorArray: string[];
+  log: typeof console.log;
+}): Promise<void> {
+  try {
+    const { clientKey, clientSecret } = await getElationClientKeyAndSecret({
+      cxId,
+      practiceId,
+    });
+    const api = await ElationApi.create({
+      practiceId,
+      environment,
+      clientKey,
+      clientSecret,
+    });
+    const appointments = await api.getAppointments({
+      cxId,
+      fromDate,
+      toDate,
+    });
+    returnArray.push(
+      ...appointments.map(appointment => {
+        return {
+          cxId,
+          elationPracticeId: practiceId,
+          elationPatientId: appointment.patient,
+        };
+      })
+    );
+  } catch (error) {
+    const cause = `Cause: ${errorToString(error)}`;
+    const details = `cxId ${cxId} practiceId ${practiceId}.`;
+    const msg = "Failed to get appointments.";
+    log(`${details} ${msg} ${cause}`);
```

Comment:
nit: not msg before details? Fine if not, just pointing out to make sure

Reply:
No longer relevant.